internal/services/mssqlmanagedinstance/mssql_managed_instance_start_stop_schedule_resource.go

```diff
@@ -0,0 +1,352 @@
+
+			properties := &schedule.StartStopManagedInstanceSchedule{
+				Properties: &schedule.StartStopManagedInstanceScheduleProperties{},
+			}
+
+			if model.Description != "" {
+				properties.Properties.Description = &model.Description
+			}
+
+			properties.Properties.ScheduleList = pointer.From(expandScheduleItemModelArray(model.ScheduleList))
+
+			if model.TimeZoneId != "" {
+				properties.Properties.TimeZoneId = &model.TimeZoneId
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			scheduleID := parse.NewManagedInstanceStartStopScheduleID(id.SubscriptionId, id.ResourceGroupName, id.ManagedInstanceName, "default")
+			metadata.SetID(scheduleID)
+
+			return nil
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedInstanceStartStopSchedulesClient
+
+			managedInstanceId, err := commonids.ParseSqlManagedInstanceID(metadata.ResourceData.Get("managed_instance_id").(string))
+			if err != nil {
+				return err
+			}
+
+			var model SqlManagedInstanceStartStopScheduleModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *managedInstanceId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *managedInstanceId, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: properties was nil", managedInstanceId)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				if model.Description != "" {
+					properties.Properties.Description = &model.Description
+				} else {
+					properties.Properties.Description = nil
+				}
+			}
+
+			if metadata.ResourceData.HasChange("schedule_list") {
+				properties.Properties.ScheduleList = pointer.From(expandScheduleItemModelArray(model.ScheduleList))
+			}
+
+			if metadata.ResourceData.HasChange("timezone_id") {
+				if model.TimeZoneId != "" {
+					properties.Properties.TimeZoneId = &model.TimeZoneId
+				} else {
+					properties.Properties.TimeZoneId = nil
+				}
+			}
+
+			properties.SystemData = nil
+
+			if _, err := client.CreateOrUpdate(ctx, *managedInstanceId, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *managedInstanceId, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedInstanceStartStopSchedulesClient
+
+			managedInstanceId, err := commonids.ParseSqlManagedInstanceID(metadata.ResourceData.Get("managed_instance_id").(string))
```

Comment:
Instead of getting the managedInstanceId from the resource data, it's recommended to parse and build it like the following codes.

```
	                id, err := parse.ManagedInstanceStartStopScheduleID(metadata.ResourceData.Id())
			if err != nil {
				return err
			}
			
			managedInstanceID := commonids.NewSqlManagedInstanceID(id.SubscriptionId, id.ResourceGroup, id.ManagedInstanceName)

```

Reply:
Updated